torch/_logging.py

```diff
@@ -0,0 +1,286 @@
+# Top level logging module for torch logging
+# Design doc: https://docs.google.com/document/d/1ZRfTWKa8eaPq1AxaiHrq4ASTPouzzlPiuquSBEJYwS8/edit#
+# Simple setup for onboarding (see above doc for more detail):
+# 1. register_log for all logs you'd like to have control of
+# 2. @loggable any classes you'd like to register as artifacts can be toggled as logged/not logged
+#    Only requirement here is that it has a __str__ method, and then instances of this class can be passed directly
+#    to log.debug(<instance here>)
+# 3. call init_logging([.. your log names .. ]) somewhere in your code
+# (before the user may attempt to set logs, but after you've executed 1. and 2.)
+
+import collections
+import itertools
+import logging
+import os
+import re
+from typing import DefaultDict, Dict, Set
+
+
+DEFAULT_LOG_LEVEL = logging.WARN
+DEFAULT_FORMATTER = logging.Formatter(
+    "[%(asctime)s] %(name)s: [%(levelname)s] %(message)s"
+)
+
+NAME_TO_LOG_NAME = {}
+NAME_TO_RECORD_TYPE = {}
```

Comment:
These records seemed to both be indexed by NAME. If so, it may be easier to understand if you create a dataclass and then have only a single map from name to blah, as this makes it evident that they both have to be set/unset at the same time.

Reply:
the keys in NAME_TO_RECORD_TYPE are a subset of the keys of NAME_TO_LOG_NAME

Reply:
Ended up doing this